scripts/rust.mts

```diff
@@ -0,0 +1,162 @@
+#!/usr/bin/env zx
+
+// Script for working with Rust projects.
+
+import 'zx/globals';
+import {
+  getCargo,
+  getToolchainArgument,
+  parseCliArguments,
+  partitionArguments,
+  popArgument,
+  workingDirectory,
+} from './helpers/utils.mts';
+
+enum Command {
+  BuildSbf = 'build-sbf',
+  Format = 'format',
+  LintClippy = 'lint-clippy',
+  LintDocs = 'lint-docs',
+  LintFeatures = 'lint-features',
+  Lint = 'lint',
+  Test = 'test',
+  Wasm = 'wasm',
+  Publish = 'publish',
+}
+
+const { command, libraryPath, args } = parseCliArguments();
+const manifestPath = path.join(libraryPath, 'Cargo.toml');
+
+async function cargo(
+  toolchain: string,
+  command: string | string[],
+  defaultArgs?: string[],
+  variables?: [string, string][]
+) {
+  const [cargoArgs, commandArgs] = partitionArguments(args, '--', defaultArgs);
+  variables?.forEach(([k, v]) => ($.env[k] = v));
+  await $`cargo ${toolchain} ${command} --manifest-path ${manifestPath} ${cargoArgs} -- ${commandArgs}`;
+}
+
+async function buildSbf() {
+  return cargo(getToolchainArgument('build'), 'build-sbf', args);
+}
+
+async function format() {
+  return cargo(
+    getToolchainArgument('format'),
+    'fmt',
+    popArgument(args, '--fix') ? [] : ['--', '--check']
+  );
+}
+
+async function lintClippy() {
+  return cargo(
+    getToolchainArgument('lint'),
+    'clippy',
+    popArgument(args, '--fix') ? ['--fix'] : []
+  );
+}
```

Comment:
Doesn't need to be done in this PR, but we typically provide some more arguments to clippy, ie. `cargo +nightly-2024-11-22 clippy --all-targets --all-features -- --deny=warnings --deny=clippy::default_trait_access --deny=clippy::arithmetic_side_effects --deny=clippy::manual_let_else --deny=clippy::used_underscore_binding`

Reply:
I try switching from `nightly-2024-08-08` to `nightly-2024-11-22` and we already get warnings on both program and client crates – I think it is better to fix those in a different PR.